pkg/commands/import/import.go

```diff
@@ -4,35 +4,41 @@
 package _import
 
 import (
+	"bytes"
+	"context"
 	"io"
 	"io/ioutil"
-	"os"
 
-	"github.com/ghodss/yaml"
-	"github.com/pkg/errors"
+	"github.com/google/go-containerregistry/pkg/authn"
 	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/client-go/dynamic"
 
-	"github.com/pivotal/build-service-cli/pkg/buildpackage"
 	"github.com/pivotal/build-service-cli/pkg/clusterstack"
 	"github.com/pivotal/build-service-cli/pkg/clusterstore"
 	"github.com/pivotal/build-service-cli/pkg/commands"
+	"github.com/pivotal/build-service-cli/pkg/config"
 	importpkg "github.com/pivotal/build-service-cli/pkg/import"
 	"github.com/pivotal/build-service-cli/pkg/k8s"
-	"github.com/pivotal/build-service-cli/pkg/lifecycle"
 	"github.com/pivotal/build-service-cli/pkg/registry"
-	"github.com/pivotal/build-service-cli/pkg/stackimage"
 )
 
 type ConfirmationProvider interface {
 	Confirm(message string, okayResponses ...string) (bool, error)
 }
 
+type Importer interface {
+	ReadDescriptor(descriptorReader io.Reader) (importpkg.DependencyDescriptor, error)
+	ImportDescriptor(ctx context.Context, keychain authn.Keychain, kpConfig config.KpConfig, descriptorReader io.Reader) error
+	ImportDescriptorDryRun(ctx context.Context, keychain authn.Keychain, kpConfig config.KpConfig, descriptorReader io.Reader) error
+	Objects() []runtime.Object
```

Comment:
Is this `Objects` still used?

Reply:
I believe its used for the --output flag

Reply:
Hmm. It seems very out of place now. The importer now has a stateful thing that has a forever growing list of objects. Can we move that to the importer or have the importer return its objects?

Reply:
ya let me play around with moving it out